### DISCUSSION TITLE
Install Vanilo Modules

### DISCUSSION THREAD

**Discussion by xujiongze** [vanilophp/framework#174] [Q&A]
I like to use Vanilo work with modules, because I have a exist project, and just need add ecommerce functions.

Im reading the documentation, the first step to install Vanilo is install framework, but I want to use module way, not framework with all modules.

So, I need install vanilo\contracts + Concord. But how ? I can't find the installation of vanilo\contracts

Is require to install framework first ?

**Comment by fulopattila122** [comment]
Which modules do you need? If you need the cart, for example, just install the module; composer will install the dependencies.

```bash
composer require vanilo/cart
```

Afterwards, edit config/concord.php and add this content:

```php
<?php

return [
    'modules' => [
        Vanilo\Cart\Providers\ModuleServiceProvider::class
    ]
];
```

To create the necessary tables, run `php artisan migrate`

Repeat the same for all the other modules you need.

---
Let's say you need the checkout and order modules as well, then:

```bash
composer require vanilo/checkout vanilo/order
```

The `config/concord.php` should look like:

```php
<?php

return [
    'modules' => [
        Vanilo\Cart\Providers\ModuleServiceProvider::class,
        Vanilo\Checkout\Providers\ModuleServiceProvider::class,
        Vanilo\Order\Providers\ModuleServiceProvider::class,
    ]
];
```

Finally run `php artisan migrate` again
